File: scripts/example_sub.py
#Python scripts must be started with the line below
#!/usr/bin/env python

import rospy
from template_v1.msg import Temp1 #import customized message

#define callback function
def callback(data):
	g_voltage_sub = data.volt1;

#define interface to voltage_sub()
def voltage_sub():

	#initialize a node with node name
	rospy.init_node('example_sub_py',anonymous=True)
	#creating a subscriber register to topic "voltage"
	#Also, whenever a message is received, call callback function
	rospy.Subscriber("voltage",Temp1,callback)

	rospy.spin()#prevent python from exit until this node is stopped

if __name__ == '__main__':
	voltage_sub()

File: src/example_sub.cpp
#include "ros/ros.h"
#include "template_v1/Temp1.h"//importing customized message

float g_voltage_sub = 0;

/*Callback function*/
void voltageCallback(const template_v1::Temp1& volt){
  g_voltage_sub = volt.volt1;
}

int main(int argc, char** argv){

  ros::init(argc, argv, "example_sub_cpp");//initialize ROS with node name
  ros::NodeHandle nh; //create a node handle
  /*Creatig a Subscriber that subscribe on topic "voltage" with buffer size 50.
    Also, it will call the callback function whenever a message is received*/
  ros::Subscriber sub = nh.subscribe("voltage",50,voltageCallback);

  ros::spin();//Enter loops that calling message callback asap

  return 0;
}
